src/tables/CPAL.js

```diff
@@ -13,5 +13,8 @@ export default new r.Struct({
   numPalettes: r.uint16,
   numColorRecords: r.uint16,
   colorRecords: new r.Pointer(r.uint32, new r.Array(ColorRecord, 'numColorRecords')),
-  colorRecordIndices: new r.Array(r.uint16, 'numPalettes')
+  colorRecordIndices: new r.Array(r.uint16, 'numPalettes'),
+  offsetPaletteTypeArray: new r.Optional(new r.Pointer(r.uint32, new r.Array(r.uint32, 'numPalettes')), t => t.version === 1),
+  offsetPaletteLabelArray: new r.Optional(new r.Pointer(r.uint32, new r.Array(r.uint16, 'numPalettes')), t => t.version === 1),
+  offsetPaletteEntryLabelArray: new r.Optional(new r.Pointer(r.uint32, new r.Array(r.uint16, 'numPaletteEntries')), t => t.version === 1)
```

Comment:
Let's use a `VersionedStruct` for this instead of `Optional` fields. e.g.

```javascript
new r.VersionedStruct(r.uint16, {
  header: {
    // current fields
  },

  0: {}, // no additional fields

  1: {
    // new fields
  }
});
```